napi-inl.h

```diff
@@ -605,6 +606,106 @@ inline Symbol::Symbol(napi_env env, napi_value value) : Name(env, value) {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// Automagic value creation
+////////////////////////////////////////////////////////////////////////////////
+
+namespace details {
+template <typename T>
+struct vf_number {
+  static Number From(napi_env env, T value) {
```

Comment:
How about an assert that `value` does not exceed 2^53? This should, however, only be done for integer `T` wider than 53 bits. On the other hand, if this check were added, it might be better off at a lower level.

Reply:
Maybe … I think it’s kind of up to the user to make sure that their values are actually representable in JS?